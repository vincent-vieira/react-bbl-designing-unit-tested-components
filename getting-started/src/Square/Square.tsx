import React from 'react';
import './Square.css';

interface SquareProps {
  playerName: string;
  onClick?: () => void;
}

const Square: React.FC<SquareProps> = ({ onClick, playerName }) => {
  return (
    <div role="cell">
      <button className="square" onClick={() => onClick && onClick()}>
        {playerName}
      </button>
    </div>
  );
};

export default Square;
