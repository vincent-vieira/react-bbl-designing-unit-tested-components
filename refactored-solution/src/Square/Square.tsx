import { FC } from 'react';
import { TicTacToePlayer } from '../useTicTacToe';
import './Square.css';

type SquareProps = {
  playerName: TicTacToePlayer;
  onClick?: () => void;
};

const Square: FC<SquareProps> = ({ onClick, playerName }) => {
  return (
    <div role="cell">
      <button className="square" onClick={() => onClick && onClick()}>
        {playerName}
      </button>
    </div>
  );
};

export default Square;
