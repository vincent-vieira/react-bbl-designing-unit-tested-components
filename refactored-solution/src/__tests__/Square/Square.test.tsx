import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Square from '../../Square/Square';

describe('Square component', () => {
  it("should display a player's name", async () => {
    render(<Square playerName="X" />);

    expect(
      await screen.findByRole('button', { name: 'X' })
    ).toBeInTheDocument();
  });

  it('should notify parent component when clicking on it', async () => {
    const handler = jest.fn();
    render(<Square playerName="X" onClick={handler} />);

    userEvent.click(await screen.findByRole('button', { name: 'X' }));

    await waitFor(() => {
      expect(handler).toHaveBeenCalledTimes(1);
    });
  });
});
