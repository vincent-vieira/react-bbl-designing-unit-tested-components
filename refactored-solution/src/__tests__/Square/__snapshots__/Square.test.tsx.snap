// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square component with player O should render properly 1`] = `
<div
  role="cell"
>
  <button
    class="square"
  >
    O
  </button>
</div>
`;

exports[`Square component with player X should render properly 1`] = `
<div
  role="cell"
>
  <button
    class="square"
  >
    X
  </button>
</div>
`;
